File: 2997-minimum-number-of-operations-to-make-array-xor-equal-to-k/2997-minimum-number-of-operations-to-make-array-xor-equal-to-k.java
class Solution {
    public int minOperations(int[] nums, int k) {
        int _xor = 0;
       for(int num : nums) _xor ^= num;

       int count = 0;
       while(k > 0 || _xor > 0) {
           if((k % 2) != (_xor % 2)) count++;
           k /= 2;
           _xor /= 2;
       }

       return count;
    }
}

File: 2997-minimum-number-of-operations-to-make-array-xor-equal-to-k/2997-minimum-number-of-operations-to-make-array-xor-equal-to-k.py
class Solution(object):
    def minOperations(self, nums, k):
        _xor, count = 0, 0
        for num in nums:
            _xor ^= num

        while k > 0 or _xor > 0:
            if (k % 2) != (_xor % 2):
                count += 1
            k //= 2
            _xor //= 2
        
        return count
        